src/cromshell/utilities/CromshellConfig.py

```diff
@@ -0,0 +1,127 @@
+import logging
+import sys
+from pathlib import Path
+
+
+class CromshellConfig:
+    """Setup Cromshell config details. Intended to be used as a singleton"""
+
+    @classmethod
+    def user_defined_slim_metadata_parameters(cls, slim_metadata_parameters_user):
+        """Override Cromwell Slim Metadata Key From Command Line"""
+        if slim_metadata_parameters_user:
+            cls.slim_metadata_parameters = (
+                slim_metadata_parameters_user
+                + "&includeKey=subWorkflowMetadata&includeKey=subWorkflowId"
+            )
+
+    @classmethod
+    def override_cromwell_config_server(cls, server_user=None, workflow_id=None):
+        """
+        Override Cromwell Server From Command Line or Environment or Submission file
+
+        This function allows users to override the cromwell server set in the
+        cromshell config file using the command line options or environment
+        variables. By default the server URL in the config file is used, however
+        if an URL is specified in the command line or environment then this
+        overrides the default. If a workflow Id is provided without a server URL
+        then the server associated with a workflow id in the submission file
+        overrides the default. If a workflow ID is provided but isn't found
+        in the submission file then the default server is used.
+        """
+
+        if server_user is None and workflow_id is None:
+            cls.LOGGER.info(
+                "Workflow id and cromwell server not specified. Using default"
+            )
+        elif server_user:
+            cls.cromwell_server = server_user
+            CromshellConfig.LOGGER.info(
+                "Cromwell server set to the provided in cli or environment argument."
+            )
+            CromshellConfig.LOGGER.info(cls.cromwell_server)
```

Comment:
You probably always want to output which server is actually being used, so just put this output once at the end of the block.

Reply:
done

Reply:
This function gets called more than once, if its in the outer loop the cromwell server gets printed regardless if there is a change in the server being used. 
```2020-11-02 09:11:59,427 cromshell.__main__                  INFO     Invoked by: /Users/bshifaw/work/cromshell2_project/cromshell/venv/bin/cromshell status 010a35db-4e36-44d0-b210-f13caa99d3ea
2020-11-02 09:11:59,427 cromshell.__main__                  INFO     Log level set to: INFO
2020-11-02 09:11:59,427 cromshell.utilities.cromshellconfig INFO     Workflow id and cromwell server not specified. Using default cromwell server
2020-11-02 09:11:59,427 cromshell.utilities.cromshellconfig INFO     Server: https://cromwell-v51.dsde-methods.broadinstitute.org
2020-11-02 09:11:59,428 cromshell.status.command            INFO     status
2020-11-02 09:11:59,428 cromshell.utilities.cromshellconfig INFO     Server: https://cromwell-v51.dsde-methods.broadinstitute.org
```
It looks redundant so i'll keep the printout in the inner loops, so that way the function can be called and if the cromwell server is not overridden then printout wont occur.